sgeop/nodes.py

```diff
@@ -172,25 +172,41 @@ def weld_edges(
     ).tolist()
 
 
-def remove_false_nodes(gdf, aggfunc="first", **kwargs):
-    """Reimplementation of momepy.remove_false_nodes that preserves attributes
+def remove_false_nodes(
+    gdf: gpd.GeoSeries | gpd.GeoDataFrame, aggfunc: str = "first", **kwargs
+):
+    """Reimplementation of ``momepy.remove_false_nodes()`` that preserves attributes.
 
     Parameters
     ----------
-    gdf : _type_
-        _description_
+    gdf : gpd.GeoSeries | gpd.GeoDataFrame
+        Input edgelines process.
+    aggfunc : str = "first"
+        Aggregate function for processing non-spatial component.
+    **kwargs
+        Keyword arguments for ``aggfunc``.
 
     Returns
     -------
-    _type_
-        _description_
+    gpd.GeoSeries | gpd.GeoDataFrame
+       The original input ``gdf`` if only 1 edgeline, otherwise the processed
+       edgeline without interstitial nodes.
     """
+
+    def merge_geometries(block: gpd.GeoSeries) -> shapely.LineString:
+        """Helper in processing the spatial component."""
+        return shapely.line_merge(shapely.GeometryCollection(block.values))
+
     if len(gdf) < 2:
         return gdf
 
     if isinstance(gdf, gpd.GeoSeries):
         gdf = gdf.to_frame("geometry")
 
+    # explode to avoid MultiLineStrings
+    # reset index due to the bug in GeoPandas explode
+    gdf = gdf.reset_index(drop=True).explode(ignore_index=True)
```

Comment:
```suggestion
    gdf = gdf.explode(ignore_index=True)
```

I don't think that workaround is no longer needed. We should also make a note in docstring that mutlilinestrings are exploded.